packages/lib-utils/src/k8s/hooks/useK8sWatchResources.test.ts

```diff
@@ -0,0 +1,160 @@
+        message: 'Application has been successfully created',
+        reason: 'OK',
+        status: 'True',
+        type: 'Created',
+      },
+    ],
+    devfile:
+      'metadata:\\n  attributes:\\n    appModelRepository.context: /\\n    appModelRepository.url: https://github.com/redhat-appstudio-appdata/-vnambiar-touch-drive\\n    gitOpsRepository.context: /\\n    gitOpsRepository.url: https://github.com/redhat-appstudio-appdata/-vnambiar-touch-drive\\nschemaVersion: 2.1.0\\n',
+  },
+};
+
+// Mock modules
+
+jest.mock('react-redux', () => ({
+  useDispatch: jest.fn(() => jest.fn()),
+  useSelector: jest.fn(),
+}));
+
+jest.mock('./k8s-watcher', () => ({
+  getWatchData: jest.fn(),
+  getReduxData: jest.fn(),
+  NoModelError: jest.fn(() => ({ message: 'Model does not exist' })),
+}));
+
+jest.mock('./useModelsLoaded', () => ({ useModelsLoaded: jest.fn() }));
+
+jest.mock('./useDeepCompareMemoize', () => ({ useDeepCompareMemoize: jest.fn() }));
+
+describe('useK8sWatchResources', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    (getWatchData as jest.Mock).mockReturnValue(null);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('should return "loaded" as false when models have not loaded', () => {
+    const haveModelsLoaded = false;
+    (useModelsLoaded as jest.Mock).mockReturnValue(haveModelsLoaded);
+    (useSelector as jest.Mock)
+      .mockReturnValueOnce(ImmutableMap<string, K8sModelCommon>()) // Models have not loaded so allK8sModels is empty
+      .mockReturnValueOnce(ImmutableMap<string, unknown>());
+    (useDeepCompareMemoize as jest.Mock).mockReturnValue(watchedResourcesMock);
+
+    const { result } = renderHook(() => useK8sWatchResources(watchedResourcesMock));
+    const { application } = result.current;
+    const { data, loaded, loadError } = application;
+
+    expect(data).toMatchObject({});
+    expect(loaded).toEqual(false);
+    expect(loadError).toBeUndefined();
+  });
+
+  test('should return specific error if the model for the watched resource does not exist', () => {
+    (useModelsLoaded as jest.Mock).mockReturnValue(true);
+    (useDeepCompareMemoize as jest.Mock).mockReturnValue(watchedResourcesMock);
+    (useSelector as jest.Mock)
+      .mockReturnValueOnce(ImmutableMap<string, K8sModelCommon>()) // Models have loaded but do not contain "application" model
+      .mockReturnValueOnce(ImmutableMap<string, unknown>());
+
+    const { result } = renderHook(() => useK8sWatchResources(watchedResourcesMock));
+    const { application } = result.current;
+    const { data, loaded, loadError } = application;
+
+    expect(data).toMatchObject({});
+    expect(loaded).toEqual(true);
+    expect((loadError as NoModelError).message).toEqual('Model does not exist');
+  });
+
+  test('should return data for the watched resource', () => {
+    const mockWatchData: WatchData = {
+      id: 'appstudio.redhat.com~v1alpha1~Application---{"ns":"test-ns","name":"test"}',
```

Comment:
It looks like this ID is shared with the payload below. Can we de-dupe it/put it in a constant?